### ISSUE TITLE
EB 4.0.0 breaks when there is another "eb" command before itself in the PATH despite being called with a full path

### ISSUE THREAD

**Issue by akesandgren** [easybuilders/easybuild-framework#3045]
Setup:
```
grep robot-paths /hpc2n/eb/easybuild.cfg 
robot-paths=/hpc2n/eb/custom/easyconfigs:%(DEFAULT_ROBOT_PATHS)s
```

/hpc2n/eb -> /cvmfs/ebsw.hpc2n.umu.se/amd64_ubuntu1604_skx/

(Last part arch dependent)

EB 3.9.4:
```
eb --show-config
robot-paths                 (F) = /hpc2n/eb/custom/easyconfigs, /hpc2n/eb/software/Core/EasyBuild/3.9.4/lib/python2.7/site-packages/easybuild_easyconfigs-3.9.4-py2.7.egg/easybuild/easyconfigs
```

EB 4.0.0:
```
eb --show-config
robot-paths                    (F) = /hpc2n/eb/custom/easyconfigs, /cvmfs/ebsw.hpc2n.umu.se/amd64_ubuntu1604_common/software/Core/EasyBuild/4.0.0/easybuild/easyconfigs
```

(Note that it also resolves the DEFAULT_ROBOT_PATHS which it shouldn't do)

```
cat /scratch/q/eb
#!/bin/bash
echo IIeeekkk, Im getting called
```

EB 3.9.4:
```
env PATH=/scratch/q:$PATH $EBROOTEASYBUILD/bin/eb --show-config
robot-paths                 (F) = /hpc2n/eb/custom/easyconfigs, /hpc2n/eb/software/Core/EasyBuild/3.9.4/lib/python2.7/site-packages/easybuild_easyconfigs-3.9.4-py2.7.egg/easybuild/easyconfigs
```

Still correct, but

EB 4.0.0:
```
env PATH=/scratch/q:$PATH $EBROOTEASYBUILD/bin/eb --show-config
robot-paths                    (F) = /hpc2n/eb/custom/easyconfigs
```

This breaks our setup completely and I'm now stuck.

**Comment by boegel** [comment]
This is related to no longer relying on `setuptools`, sort of...

The location of the `eb` command is used as a fallback to determine the location of the easyconfig files included with EasyBuild (that's done in `get_paths_for` from `easybuild.framework.easyconfigs.tools`).

When EasyBuild is installed using `setuptools` (which you can still do via `easy_install` for example), the fallback based on the location to the `eb` command is not used (because the location to the easyconfig files is included in `$PYTHONPATH`).

If it's not, the location of the easyconfig files is assumed to be determined by the location of `eb`; so, if `eb` is in `<prefix>/bin/eb`, then the easyconfig files can be found in `<prefix>/easybuild/easyconfigs/`.

Currently the `eb` script doesn't pass its own location to `easybuild.main`, but it could (by setting an environment variable that gets picked up for example, in which case we don't need to do a `which eb` at all.

Resolving the path to `eb` is done for good reason, see the fix in #2248.
We could consider both the original path and the resolved path though, so that's fairly easy to fix...

In short: it shouldn't be too difficult to rectify this, and not relying on `$PATH` to find the easyconfigs included with EasyBuild at all is probably better in general...

@akesandgren For the sake of context, how exactly did EasyBuild 4.0.0 get installed in your setup?

**Comment by akesandgren** [comment]
Ahh, 4.0.0 was installed with 3.9.4 using --install-latest-eb-release

**Comment by boegel** [comment]
proposed fix in #3046

**Comment by akesandgren** [comment]
Fix works, closing
